hack/chains/kaniko-cosign-verify.sh

```diff
@@ -11,11 +10,14 @@ TASKRUN_NAME=taskrun/$( trim-name $TASKRUN_NAME )
 IMAGE_URL=$( oc get $TASKRUN_NAME -o json | jq -r '.status.taskResults[1].value' )
 IMAGE_REGISTRY=$( echo $IMAGE_URL | cut -d/ -f1 )
 #IMAGE_REGISTRY=$( oc registry info )
+INTERNAL_REGISTRY=$([[ $IMAGE_REGISTRY == 'image-registry.openshift-image-registry.svc:5000' ]] && echo true || echo false)
 
-title "Make sure we're logged in to the registry"
-# Make sure we have a docker credential since cosign will need it
-# (Todo: Probably shouldn't assume kubeadmin user here)
-oc whoami -t | docker login -u kubeadmin --password-stdin $IMAGE_REGISTRY
+if ! $INTERNAL_REGISTRY; then
+  title "Make sure we're logged in to the registry"
+  # Make sure we have a docker credential since cosign will need it
+  # (Todo: Probably shouldn't assume kubeadmin user here)
+  oc whoami -t | docker login -u kubeadmin --password-stdin $IMAGE_REGISTRY
```

Comment:
If the registry from the image reference is internal this command will fail since local machine can't access via service, i.e. would get:

```
Error response from daemon: Get "https://image-registry.openshift-image-registry.svc/v2/": dial tcp: lookup image-registry.openshift-image-registry.svc: no such host
```

That's why we pivot to running the task in-cluster below.

Reply:
That's not what I meant :)

I'm was talking about the `Todo` comment. You don't have to provide `-u kubeadmin`:
```
oc whoami -t | docker login -u unused --password-stdin $IMAGE_REGISTRY
```

Reply:
I get it now, I'll create a followup PR, this wasn't changed in this PR so best kept it as a separate followup cleanup PR.